### PR TITLE
Test All-In-One packer template on a DigitalOcean agent (JDK8)

### PR DESCRIPTION
**Pull request by smerle33**
to provide test agent on ci.jenkins.io for the AllInOne template from packer

### CODE REVIEW

File: hieradata/clients/azure.ci.jenkins.io.yaml [hunk 1]
```diff
@@ -267,6 +267,13 @@ profile::jenkinscontroller::jcasc:
             cpus: 4
             memory: 8
             imagePullSecrets: dockerhub-credential
+          - name: jnlp-maven-8-aio
```
Comment (lemeurherve) [inline hieradata/clients/azure.ci.jenkins.io.yaml]:
```suggestion
          - name: jnlp-maven-all-in-one
```

File: hieradata/common.yaml [hunk 2]
```diff
@@ -219,6 +219,7 @@ profile::jenkinscontroller::jcasc:
       jnlp-maven-11-windows: jenkinsciinfra/inbound-agent-maven:jdk11-nanoserver
       jnlp-ruby: jenkinsciinfra/inbound-agent-ruby@sha256:221f96baa1957742728eb6d2769a691aea5721fc11ed9b05da810c5debe92115
       jnlp-maven-8: jenkinsciinfra/inbound-agent-maven@sha256:4082d59973b7686af97dd4847c46b8affb57e1742820cc1fc238f847efc85f3a
+      jnlp-maven-8-aio: jenkinsciinfra/jenkins-agent-ubuntu-20.04@sha256:3a490163d60ff134481964b816ecac919b7b8699189d86e0db634b8d36678306
```
Comment (lemeurherve) [inline hieradata/common.yaml]:
```suggestion
     jnlp-maven-8-all-in-one: jenkinsciinfra/jenkins-agent-ubuntu-20.04@sha256:3a490163d60ff134481964b816ecac919b7b8699189d86e0db634b8d36678306
```

File: hieradata/clients/azure.ci.jenkins.io.yaml (hunk 1 above)
Comment (lemeurherve) [reply hieradata/clients/azure.ci.jenkins.io.yaml]: I would either put this all-in-one before other jnlp-maven-* agent definitions, or either after them (l285)

Comment (smerle33) [reply hieradata/clients/azure.ci.jenkins.io.yaml]: I disagree, the 8 mean that the default path for javahome on that agent will be on jdk 8, and I will build agent for 11 and 17 with the exact same image but with a specific javahome path line 276 so the agent name must be specific also to help choosing the correct one. 
But I can change 'aio' by 'all-in-one'

File: hieradata/common.yaml (hunk 2 above)
Comment (smerle33) [reply hieradata/common.yaml]: for now, the template is forcing the same image name and the same agent name https://github.com/jenkins-infra/jenkins-infra/blob/b1657c5bef338d51be415f69f6d925533ac00e06/dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb#L199 so it need to contain the 8, 11 and 17

Comment (lemeurherve) [reply hieradata/common.yaml]: I've updated my suggestion accordingly.

File: dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb [hunk 3]
```diff
@@ -205,6 +205,14 @@ jenkins:
               - envVar:
                   key: "JENKINS_JAVA_BIN"
                   value: "<%= agent['agentJavaBin'] ? agent['agentJavaBin'] : @jcasc['agents_setup']['ubuntu']['agentJavaBin'] %>"
+            <%- if agent['javahome'] -%>
+              - envVar:
+                  key: "JAVA_HOME"
+                  value: "<%= agent['javahome'] %>"
+              - envVar:
+                  key: "PATH+JAVAPATH"
+                  value: "<%= agent['javahome'] %>/bin"
```
Comment (lemeurherve) [inline dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb]:
```suggestion
                  value: "<%= agent['javaPath'] %>"
              - envVar:
                  key: "PATH+JAVAPATH"
                  value: "<%= agent['javaPath'] %>/bin"
```

File: hieradata/clients/azure.ci.jenkins.io.yaml [hunk 4]
```diff
@@ -267,6 +267,13 @@ profile::jenkinscontroller::jcasc:
             cpus: 4
             memory: 8
             imagePullSecrets: dockerhub-credential
+          - name: jnlp-maven-8-all-in-one
+            labels:
+              - maven-8-all-in-one
+            cpus: 4
+            memory: 8
+            imagePullSecrets: dockerhub-credential
+            javahome: /opt/jdk-8
```
Comment (lemeurherve) [inline hieradata/clients/azure.ci.jenkins.io.yaml]:
```suggestion
            javaHome: /opt/jdk-8
```

File: dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb [hunk 5]
```diff
@@ -205,6 +205,14 @@ jenkins:
               - envVar:
                   key: "JENKINS_JAVA_BIN"
                   value: "<%= agent['agentJavaBin'] ? agent['agentJavaBin'] : @jcasc['agents_setup']['ubuntu']['agentJavaBin'] %>"
+            <%- if agent['javahome'] -%>
+              - envVar:
+                  key: "JAVA_HOME"
+                  value: "<%= agent['javahome'] %>"
+              - envVar:
+                  key: "PATH+JAVAPATH"
```
Comment (lemeurherve) [inline dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb]:
```suggestion
                  key: "PATH+JAVA_HOME"
```

**Review by lemeurherve — COMMENTED** [review]
Some naming suggestions (again, sorry ^^)

File: dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb [hunk 6]
```diff
@@ -205,6 +205,14 @@ jenkins:
               - envVar:
                   key: "JENKINS_JAVA_BIN"
                   value: "<%= agent['agentJavaBin'] ? agent['agentJavaBin'] : @jcasc['agents_setup']['ubuntu']['agentJavaBin'] %>"
+            <%- if agent['javahome'] -%>
+              - envVar:
+                  key: "JAVA_HOME"
+                  value: "<%= agent['javahome'] %>"
+              - envVar:
+                  key: "PATH+JAVA_HOME"
+                  value: "<%= agent['javahome'] %>/bin"
```
Comment (lemeurherve) [inline dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb]:
```suggestion
                  value: "<%= agent['javaHome'] %>"
              - envVar:
                  key: "PATH+JAVA_HOME"
                  value: "<%= agent['javaHome'] %>/bin"
```

File: hieradata/clients/azure.ci.jenkins.io.yaml (hunk 4 above)
Comment (smerle33) [reply hieradata/clients/azure.ci.jenkins.io.yaml]: `            javaHome: /opt/jdk-8`
shouldn't it be javaPath to match the templating ? 
`value: "<%= agent['javaPath'] %>"`

**Review by lemeurherve — APPROVED** [review]

File: dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb [hunk 7]
```diff
@@ -205,6 +205,14 @@ jenkins:
               - envVar:
                   key: "JENKINS_JAVA_BIN"
                   value: "<%= agent['agentJavaBin'] ? agent['agentJavaBin'] : @jcasc['agents_setup']['ubuntu']['agentJavaBin'] %>"
+            <%- if agent['javahome'] -%>
```
Comment (smerle33) [inline dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb:208]: shouldn't it be javaHome ????

Comment (lemeurherve) [reply dist/profile/templates/jenkinscontroller/casc/clouds.yaml.erb:208]: You're right 😓 
Creating a PR to fix it

**Pull request merged**